src/include/distributed/multi_executor.h

```diff
@@ -74,13 +74,26 @@ extern void CitusExecutorRun(QueryDesc *queryDesc, ScanDirection direction, uint
 							 bool execute_once);
 extern void AdaptiveExecutorPreExecutorRun(CitusScanState *scanState);
 extern TupleTableSlot * AdaptiveExecutor(CitusScanState *scanState);
-extern uint64 ExecuteTaskListExtended(RowModifyLevel modLevel, List *taskList,
-									  TupleDesc tupleDescriptor,
-									  Tuplestorestate *tupleStore,
-									  bool hasReturning, int targetPoolSize,
-									  TransactionProperties *xactProperties,
-									  List *jobIdList,
-									  bool localExecutionSupported);
+
+typedef struct ExecutionParams
+{
+	RowModifyLevel modLevel;
+	List *taskList;
+	TupleDesc tupleDescriptor;
+	Tuplestorestate *tupleStore;
+	bool hasReturning;
+	int targetPoolSize;
+	TransactionProperties *xactProperties;
```

Comment:
Maybe just inline this.

Reply:
inline this as in remove TransactionProperties and use it directly here?

Reply:
Unrelated: This should be in adaptive_executor.h ideally. We should probably move the related functions to `adaptive_executor.h` from `multi_executor.h`.

Reply:
As in, use a TransactionProperties struct instead of a pointer to a struct.